.github/workflows/pipelines-unlock.yml

```diff
@@ -0,0 +1,83 @@
+name: Pipelines
+run-name: Run Gruntwork Pipelines
+on:
+  workflow_call:
+    inputs:
+      lock_id:
+        description: "The ID of the lock, usually a GUID"
+        required: true
+        type: string
+      working_directory:
+        description: "Path to the terragrunt.hcl file where the lock is held"
+        required: true
+        type: string
+
+      # This field can be overriden to customize the runner used for pipelines
+      # workflows.
+      #
+      # IMPORTANT: To use self-hosted runners this workflow must be hosted in
+      # the same GitHub organization as your infra-live repository.
+      # See https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-self-hosted-runners
+      #
+      # The value must be an escaped JSON string that will be decoded to the
+      # jobs.runs-on field
+      # See https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idruns-on
+      #
+      # For example:
+      # - A simple github runner: "\"ubuntu-22.04\""
+      # - A list of labels: "[\"self-hosted\", \"linux\"]"
+      # - A map: "{group: \"ubuntu-runners\", labels: \"ubuntu-20.04-16core\"}"
+      runner:
+        type: string
+        default: '"ubuntu-latest"'
+    secrets:
+        PIPELINES_READ_TOKEN:
+          required: true
+env:
+  PIPELINES_CLI_VERSION: v0.20.0
+  PIPELINES_ACTIONS_VERSION: event
+  GITHUB_EVENT_NAME: push
+
+jobs:
+  unlock:
+    name: Unlock TF State
+    runs-on: ${{ fromJSON(inputs.runner) }}
+    steps:
+      - name: Checkout Pipelines Actions
+        uses: actions/checkout@v4
+        with:
+          path: pipelines-actions
+          repository: gruntwork-io/pipelines-actions
+          ref: ${{ env.PIPELINES_ACTIONS_VERSION }}
+          token: ${{ secrets.PIPELINES_READ_TOKEN }}
+
+      - name: Check out repo code
+        uses: actions/checkout@v4
+        with:
+          path: infra-live-repo
+          fetch-depth: 0
+
+      - name: Bootstrap Workflow
+        id: gruntwork_context
+        uses: ./pipelines-actions/.github/actions/pipelines-bootstrap
+        with:
+          token: ${{ secrets.PIPELINES_READ_TOKEN }}
+          working_directory: ${{ inputs.working_directory }}
+          terragrunt_command: "force-unlock -force ${{ inputs.lock_id }}"
+          branch: 'main'
+
+      - name: "Run terragrunt force-unlock in ${{ inputs.working_directory }}"
+        id: terragrunt
+        uses: ./pipelines-actions/.github/actions/pipelines-execute
+        env:
+          TERRAGRUNT_AUTH_PROVIDER_CMD: "pipelines auth terragrunt-credentials --ci github-actions --cloud aws --wd ."
+        with:
+          token: ${{ secrets.PIPELINES_READ_TOKEN }}
+          tf_binary: ${{ steps.gruntwork_context.outputs.tf_binary }}
+          working_directory: ${{ steps.gruntwork_context.outputs.working_directory }}
+          terragrunt_command: ${{ steps.gruntwork_context.outputs.terragrunt_command }}
```

Comment:
Should we be setting the terragrunt "force unlock" command here or is that happening in bootstrap?

Reply:
You're correct, that's happening in bootstrap and it gets carried forwards